smoothscroll.js

```diff
@@ -1,156 +1,237 @@
-(function () {
+(function (w, doc, undefined) {
   'use strict';
 
-  if ('scrollBehavior' in document.documentElement.style) return;
-
-  // TODO: make this intelligent according to distance.
-  var SCROLL_TIME = 300;
-
-  var originalScrollTo = window.scrollTo;
-  var originalScrollBy = window.scrollBy;
-  var originalScrollIntoView = Element.prototype.scrollIntoView;
-
-  // store generally accessible frame id in case a new scroll animation is triggered before the previous
-  // completes, we can cancel the previous scroll.
-  var frame;
-
-  var startY, startX, endX, endY;
-
+  /*
+   * alias
+   * w: window global object
+   * doc: document
+   * undefined: undefined
+   */
+
+  // return if scrollBehavior is supported
+  if ('scrollBehavior' in doc.documentElement.style) return;
+
+  var SCROLL_TIME = 400,
+      // legacy scrolling methods
+      originalScrollTo = w.scrollTo,
+      originalScrollBy = w.scrollBy,
+      originalScrollIntoView = w.Element.prototype.scrollIntoView,
+      // global frame variable to avoid collision
+      frame,
+      // global metric variables
+      startY, startX, endX, endY;
+
+  /*
+   * returns actual time
+   * @method now
+   * @returns {Date}
+   */
   function now() {
-    return window.performance !== undefined && window.performance.now !== undefined ? window.performance.now() : Date.now !== undefined ? Date.now() : new Date().getTime();
+    return w.performance !== undefined && w.performance.now !== undefined ?
+      // if performance object supported return now, if not fallback to date object
+      w.performance.now() : Date.now !== undefined ? Date.now() : new Date().getTime();
   }
 
-  // ease-in-out
+  /*
+   * returns result of applying ease math function to a number
+   * @method ease
+   * @param {Number} k
+   * @returns {Number}
+   */
   function ease(k) {
     return 0.5 * (1 - Math.cos(Math.PI * k));
   }
 
-  function shouldBailOut(x) {
-    return typeof x === 'undefined' || x.behavior !== 'smooth';
+  /*
+   * returns true if first argument is an options object and contains a smooth behavior
+   * @method bailOut
+   * @param {Number|Object} x
+   * @returns {Boolean}
+   */
+  function bailOut(x) {
```

Comment:
nit: I still prefer the `should` prefix since this is only called at the beginning of one of the scroll methods to carry on with the polyfill or bail out to the original method.